Tests/SwiftCurrentTests/CustomAssertions.swift

```diff
@@ -11,12 +11,16 @@ import XCTest
 
 @testable import CwlPreconditionTesting
 
-func XCTAssertThrowsFatalError(instructions: @escaping () -> Void, file: StaticString = #file, line: UInt = #line) {
+func XCTAssertThrowsFatalError(instructions: @escaping () -> Void, file: StaticString = #file, line: UInt = #line) throws {
+    #if (os(macOS) || os(iOS)) && arch(x86_64)
     var reached = false
     let exception = catchBadInstruction {
         instructions()
         reached = true
     }
     XCTAssertNotNil(exception, "No fatal error thrown", file: file, line: line)
     XCTAssertFalse(reached, "Code executed past expected fatal error", file: file, line: line)
+    #else
+    throw XCTSkip("XCTAssertThrowsFatalError is only available on macOS/iOS on x86_64 architecture.")
+    #endif
 }
```

Comment:
I think we can be a little more explicit and a lot more flexible with something like this.

```suggestion
func XCTAssertThrowsFatalError(instructions: @escaping () -> Void, file: StaticString = #file, line: UInt = #line) throws {
    try XCTSkipUnless(ProcessInfo.canCatchBadInstruction, "XCTAssertThrowsFatalError is only available on macOS/iOS on x86_64 architecture.")
    var reached = false
    let exception = catchBadInstruction {
        instructions()
        reached = true
    }
    XCTAssertNotNil(exception, "No fatal error thrown", file: file, line: line)
    XCTAssertFalse(reached, "Code executed past expected fatal error", file: file, line: line)
}

extension ProcessInfo {
    static var isx86: Bool {
        #if arch(x86_64)
            return true
        #else
            return false
        #endif
    }

    static var ismacOS: Bool {
        #if os(macOS)
            return true
        #else
            return false
        #endif
    }

    static var isiOS: Bool {
        #if os(iOS)
            return true
        #else
            return false
        #endif
    }

    fileprivate static var canCatchBadInstruction: Bool {
        (ismacOS || isiOS) && isx86
    }
}
```

Reply:
As an added benefit making more things runtime like this means it's harder for somebody to accidentally break stuff when running on a particular architecture.

Reply:
To prove out my previous statement on your M1 mac go delete a character from inside the `#if` but before the `#else`. It'll still happily compile for you, but not anybody running x86.

Reply:
The issue that mine is solving is that it's a compile-time error.  My computer doesn't have `catchBadInstruction` to even reference, so I can't compile.  Otherwise, I would agree that not borking the other architectures would be ideal.